Mathlib/CategoryTheory/Sites/Coherent/RegularSheaves.lean

```diff
@@ -54,6 +54,147 @@ def EqualizerCondition (P : Cᵒᵖ ⥤ Type*) : Prop :=
+    simp only [Functor.comp_obj, Functor.op_obj, unop_op, Functor.comp_map, Functor.op_map,
+      Quiver.Hom.unop_op]
+    have := hasPullback_of_preservesPullback F π π
+    have h := equalizer.condition (P.map (pullback.fst (f := (F.map π)) (g := (F.map π))).op)
+      (P.map (pullback.snd).op)
+    simp only [← PreservesPullback.iso_hom_fst, ← PreservesPullback.iso_hom_snd, op_comp,
+      Functor.map_comp, ← Category.assoc, h])
+  inv := equalizer.lift (equalizer.ι _ _) (by
+    simp only [Functor.comp_obj, Functor.op_obj, unop_op, Functor.comp_map, Functor.op_map,
+      Quiver.Hom.unop_op]
+    have := hasPullback_of_preservesPullback F π π
+    simp only [← PreservesPullback.iso_inv_fst, ← PreservesPullback.iso_inv_snd,
+      op_comp, Functor.map_comp]
+    rw [← Category.assoc, equalizer.condition]
+    rfl)
+  hom_inv_id := by
+    apply equalizer.hom_ext
+    simp only [Functor.comp_obj, Functor.op_obj, unop_op, Functor.comp_map, Functor.op_map,
+      Quiver.Hom.unop_op, Category.assoc, limit.lift_π, op_comp, id_eq, eq_mpr_eq_cast, cast_eq,
+      Fork.ofι_pt, Fork.ofι_π_app, Category.id_comp]
+    exact equalizer.lift_ι _ _
+  inv_hom_id := by
+    apply equalizer.hom_ext
+    simp only [Functor.comp_obj, Functor.op_obj, unop_op, Functor.comp_map, Functor.op_map,
+      Quiver.Hom.unop_op, Category.assoc, Category.id_comp]
+    erw [equalizer.lift_ι, equalizer.lift_ι]
+
+theorem mapToEqualizer_pullback_comp {D : Type*} [Category D] (P : Cᵒᵖ ⥤ Type*) (F : D ⥤ C)
+    {X B : D} (π : X ⟶ B) [HasPullback π π] [PreservesLimit (cospan π π) F] :
+    haveI := hasPullback_of_preservesPullback F π π
+    (equalizer.lift ((F.op ⋙ P).map π.op)
+      (equalizerCondition_iff_isIso_lift_w (F.op ⋙ P) π)) =
+    (equalizer.lift (P.map (F.map π).op)
+      (equalizerCondition_iff_isIso_lift_w P (F.map π))) ≫
+        (mapToEqualizer_pullback_comp_aux P F π).hom := by
+  simp only [Functor.comp_obj, Functor.op_obj, unop_op, Functor.comp_map, Functor.op_map,
+    Quiver.Hom.unop_op, mapToEqualizer_pullback_comp_aux]
+  apply equalizer.hom_ext
+  simp only [Functor.comp_obj, Functor.op_obj, unop_op, Functor.comp_map, Functor.op_map,
+    Quiver.Hom.unop_op, Category.assoc]
+  erw [equalizer.lift_ι, equalizer.lift_ι, equalizer.lift_ι]
+
+theorem equalizerCondition_precomp_of_preservesPullback {D : Type*} [Category D] (P : Cᵒᵖ ⥤ Type*)
+    (F : D ⥤ C)
+    [∀ {X B} (π : X ⟶ B) [EffectiveEpi π], PreservesLimit (cospan π π) F]
+    [F.PreservesEffectiveEpis] -- merge the stuff from the effective epi file added in #10013
+    (hP : EqualizerCondition P) : EqualizerCondition (F.op ⋙ P) := by
+  rw [equalizerCondition_iff_isIso_lift] at hP ⊢
+  intro X B π _ _
+  have := hasPullback_of_preservesPullback F π π
+  have := hP (F.obj X) (F.obj B) (F.map π)
+  rw [mapToEqualizer_pullback_comp (F := F)]
+  exact IsIso.comp_isIso
+
+theorem equalizerCondition_of_natIso_aux
+    {P : Cᵒᵖ ⥤ Type*} {P' : Cᵒᵖ ⥤ Type _} (i : P ≅ P') {X B : C} (π : X ⟶ B)
```

Comment:
```suggestion
    {P P' : Cᵒᵖ ⥤ Type u} (i : P ≅ P') {X B : C} (π : X ⟶ B)
```
This is exactly the same (declaring `u` of course), but I think it is more readable. (Same below.)